tests/test-rest-posts-terms-controller.php

```diff
@@ -0,0 +1,235 @@
+
+		wp_set_current_user( $this->admin_id );
+		$request = new WP_REST_Request( 'POST', sprintf( '/wp/v2/posts/%d/terms/post_tag/%d', $this->post_id, 9999 ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_term_invalid', $response, 404 );
+	}
+
+	public function test_delete_item() {
+		wp_set_current_user( $this->admin_id );
+		$tag = wp_insert_term( 'test-tag', 'post_tag' );
+		wp_set_object_terms( $this->post_id, 'test-tag', 'post_tag' );
+
+		$request = new WP_REST_Request( 'DELETE', sprintf( '/wp/v2/posts/%d/terms/post_tag/%d', $this->post_id, $tag['term_taxonomy_id'] ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertFalse( is_object_in_term( $this->post_id, 'post_tag', $tag['term_id'] ) );
+	}
+
+	public function test_delete_item_invalid_permission() {
+		$tag = wp_insert_term( 'test-tag', 'post_tag' );
+		wp_set_object_terms( $this->post_id, 'test-tag', 'post_tag' );
+
+		$request = new WP_REST_Request( 'DELETE', sprintf( '/wp/v2/posts/%d/terms/post_tag/%d', $this->post_id, $tag['term_taxonomy_id'] ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_forbidden', $response, 403 );
+	}
+
+	public function test_delete_item_invalid_taxonomy() {
+		wp_set_current_user( $this->admin_id );
+		$tag = wp_insert_term( 'test-tag', 'post_tag' );
+		wp_set_object_terms( $this->post_id, 'test-tag', 'post_tag' );
+
+		$request = new WP_REST_Request( 'DELETE', sprintf( '/wp/v2/posts/%d/terms/invalid_taxonomy/%d', $this->post_id, $tag['term_taxonomy_id'] ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_no_route', $response, 404 );
+	}
+
+	public function test_delete_item_invalid_taxonomy_term() {
+		wp_set_current_user( $this->admin_id );
+		$tag = wp_insert_term( 'test-tag', 'post_tag' );
+		wp_set_object_terms( $this->post_id, 'test-tag', 'post_tag' );
+
+		$request = new WP_REST_Request( 'DELETE', sprintf( '/wp/v2/posts/%d/terms/invalid_taxonomy/%d', $this->post_id, $tag['term_taxonomy_id'] ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_no_route', $response, 404 );
+	}
+
+	public function test_delete_item_invalid_post() {
+		wp_set_current_user( $this->admin_id );
+		$tag = wp_insert_term( 'test-tag', 'post_tag' );
+		wp_set_object_terms( $this->post_id, 'test-tag', 'post_tag' );
+
+		$request = new WP_REST_Request( 'DELETE', sprintf( '/wp/v2/posts/%d/terms/post_tag/%d', 9999, $tag['term_taxonomy_id'] ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_post_invalid_id', $response, 404 );
+	}
+
+	public function test_update_item() {
+		/** Can't update a relationship **/
+	}
+
+	public function test_prepare_item() {
+		/** Post types can't be updated **/
+	}
+
+	public function test_get_item_schema() {
+		/** Post types can't be deleted **/
```

Comment:
Implemented test and added the schema endpoint in d3d6d6e3b2bc0f99f3de38f0bd2a84093a4d8ab8